test/unit/math/rev/mat/vectorize/apply_scalar_unary_test.cpp

```diff
@@ -0,0 +1,18 @@
+#include <stan/math/rev/core/var.hpp>
+#include <stan/math/rev/scal/fun/exp.hpp>
+#include <stan/math/rev/mat/vectorize/apply_scalar_unary.hpp>
+#include <test/unit/math/prim/mat/vectorize/foo_base_test.hpp>
+#include <test/unit/math/prim/mat/vectorize/expect_types.hpp>
+#include <test/unit/math/rev/mat/vectorize/expect_values.hpp>
+#include <test/unit/math/rev/mat/vectorize/expect_errors.hpp>
+#include <gtest/gtest.h>
+
+/**
+ * this tests that the expect_values test works on a mock function 
+ */
+TEST(MathRevMatVectorize, applyScalarUnaryMock) {
```

Comment:
And because it's a function and we're not allowing C++11 yet,
you can't partially specialize a function.  So I'd go with
just renaming them.

- Bob

> On Mar 4, 2016, at 5:46 PM, rayleigh <notifications@github.com> wrote:
> 
> In test/unit/math/rev/mat/vectorize/apply_scalar_unary_test.cpp:
> 
> > @@ -0,0 +1,18 @@
> > +#include <stan/math/rev/core/var.hpp>
> > +#include <stan/math/rev/scal/fun/exp.hpp>
> > +#include <stan/math/rev/mat/vectorize/apply_scalar_unary.hpp>
> > +#include <test/unit/math/prim/mat/vectorize/foo_base_test.hpp>
> > +#include <test/unit/math/prim/mat/vectorize/expect_types.hpp>
> > +#include <test/unit/math/rev/mat/vectorize/expect_values.hpp>
> > +#include <test/unit/math/rev/mat/vectorize/expect_errors.hpp>
> > +#include <gtest/gtest.h>
> > +
> > +/**
> > + * this tests that the expect_values test works on a mock function 
> > + */
> > +TEST(MathRevMatVectorize, applyScalarUnaryMock) {
> 
> My issue is the second case that you mentioned, Bob, because take the function signatures for expect_scalar_value():
> 
> In test/unit/math/prim/mat/vectorize/expect_scalar_value.hpp:
> 
> template <typename F>
> void expect_scalar_value() {
> 
> In test/unit/math/rev/mat/vectorize/expect_scalar_value.hpp:
> 
> template <typename F>
> void expect_scalar_value() {
> 
> In test/unit/math/fwd/mat/vectorize/expect_scalar_value.hpp:
> 
> template <typename F, typename T>
> void expect_scalar_value() {
> 
> In test/unit/math/mix/mat/vectorize/expect_scalar_value.hpp:
> 
> template <typename F, typename T>
> void expect_scalar_value() {
> 
> I'm not sure where I can add more specification without adding a dummy parameter/argument of some sort.
> 
> —
> Reply to this email directly or view it on GitHub.
>